src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/test_role_commands_thru_mock.py

```diff
@@ -157,10 +157,52 @@ def mock_app_create(parameters):
         self.assertTrue(logger_mock.warning.called)  # we should warn 'years' will be dropped
         self.assertTrue(faked_graph_client.applications.create.called)
 
+    @mock.patch('azure.cli.command_modules.role.custom._auth_client_factory', autospec=True)
+    @mock.patch('azure.cli.command_modules.role.custom._graph_client_factory', autospec=True)
+    def test_create_for_rbac_failed_with_polished_error_if_due_to_permission(self, graph_client_mock, auth_client_mock):
+
+        def _test_deserializer(resp_type, response):
+            err = FakedError('Insufficient privileges to complete the operation')
+            return err
+
+        faked_role_client = mock.MagicMock()
+        faked_role_client.config.subscription_id = self.subscription_id
+        auth_client_mock.return_value = faked_role_client
+        faked_graph_client = mock.MagicMock()
+        graph_client_mock.return_value = faked_graph_client
+
+        faked_graph_client.applications.create.side_effect = GraphErrorException(_test_deserializer, None)
+
+        # action
+        with self.assertRaises(CLIError) as context:
+            create_service_principal_for_rbac('will-fail', skip_assignment=True)
+
+        # assert we handled such error
+        self.assertTrue('https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal' in str(context.exception))
+
+    @mock.patch('azure.cli.command_modules.role.custom._auth_client_factory', autospec=True)
+    @mock.patch('azure.cli.command_modules.role.custom._graph_client_factory', autospec=True)
+    def test_create_for_rbac_failed_with_regular_error(self, graph_client_mock, auth_client_mock):
+
+        def _test_deserializer(resp_type, response):
+            err = FakedError('something bad for you')
+            return err
+
+        faked_role_client = mock.MagicMock()
+        faked_role_client.config.subscription_id = self.subscription_id
+        auth_client_mock.return_value = faked_role_client
+        faked_graph_client = mock.MagicMock()
+        graph_client_mock.return_value = faked_graph_client
+
```

Comment:
It looks like there's some duplicate code in these two tests, like the `_test_deserializer` code.

Reply:
Indeed, let me get rid of